### PR TITLE
Feat/more scaffolding

### PR DESCRIPTION
**Pull request by jrhart08**
net contribution of -127,781 LOC

### CODE REVIEW

File: Speakers.UI/src/nodash/renameKeys.spec.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/* eslint-env jest */
+import renameKeys from './renameKeys';
+
+describe('renameKeys', () => {
```
Comment (devzeebo) [inline Speakers.UI/src/nodash/renameKeys.spec.js]: Rewrite using `jest-gwt`

File: Speakers.UI/src/redux/reducers/themes/index.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+const initialState = {
+  current: 'dark',
+};
+
+const handlers = {
+  [THEME_CHANGED]: onThemeChanged,
+};
+
+export default (state = initialState, action) => {
```
Comment (devzeebo) [inline Speakers.UI/src/redux/reducers/themes/index.js]: use `reduceReducers`

File: Speakers.UI/src/redux/reducers/themes/onThemeChanged.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export default (state, { theme }) => ({
```
Comment (devzeebo) [inline Speakers.UI/src/redux/reducers/themes/onThemeChanged.js]: use `reducerOfType`

File: Speakers.UI/src/App.js [hunk 4]
```diff
@@ -1,30 +1,15 @@
...
-import CssBaseline from '@material-ui/core/CssBaseline';
-import Container from '@material-ui/core/Container';
 
-import theme from './theme';
-// import { Route } from 'react-router';
-import Header from './components/Header';
-import Footer from './components/Footer';
-import EngagementsList from './pages/engagements-list';
+import store from './redux/store';
+import AppInner from './App.inner';
```
Comment (devzeebo) [inline Speakers.UI/src/App.js]: Why is the Redux provider and BrowserRouter not part of the Inner, but the Mui Theme provider and Styled Theme provider are?

File: Speakers.UI/src/App.inner.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+      type: current,
+    },
+  }),
+);
+
+const App = () => {
+  const theme = useSelector(selectTheme);
+
+  return (
+    <ThemeProvider theme={theme}>
```
Comment (devzeebo) [inline Speakers.UI/src/App.inner.js]: Can we make a combined theme provider that extracts out the `theme` stuff so we don't have all this stuck in the main app file

File: Speakers.UI/src/nodash/renameKeys.spec.js (hunk 1 above)
Comment (jrhart08) [reply Speakers.UI/src/nodash/renameKeys.spec.js]: I can add GWT scenarios but for basic functions like this I still find it useful to see how to actually use the function

File: Speakers.UI/src/App.inner.js (hunk 5 above)
Comment (jrhart08) [reply Speakers.UI/src/App.inner.js]: yeah I like that idea. that lets us avoid an AppInner component too

File: Speakers.UI/src/redux/reducers/themes/index.js (hunk 2 above)
Comment (jrhart08) [reply Speakers.UI/src/redux/reducers/themes/index.js]: I'll do you one better and just provide a `useThemeContext` hook

File: Speakers.UI/src/nodash/renameKeys.spec.js (hunk 1 above)
Comment (devzeebo) [reply Speakers.UI/src/nodash/renameKeys.spec.js]: Sure, but you can click through to the definitions:

```
test('rename keys', {
  given: {
    an_object,
    new_key_mapping,
  },
  when: {
    remapping_keys,
  },
  then: {
    original_object_NOT_modified,
    new_object_has_NEW_keys,
    new_object_has_UNMAPPED_keys,
  },
});

test('ignores extra key mappings', {
  given: {
    an_object,
    new_key_mapping_WITH_EXTRA_KEYS,
  },
  when: {
    remapping_keys,
  },
  then: {
    original_object_NOT_modified,
    new_object_has_NEW_keys,
    new_object_has_UNMAPPED_keys,
    new_object_DOES_NOT_have_EXTRA_keys,
  },
});
```

**Pull request merged**